### PR TITLE
Combining issue 19 through 22

### PR DESCRIPTION
**Pull request by SergejMuhic**
https://github.com/AECgeeks/infra-repo-issue-test-1/issues/19

to

https://github.com/AECgeeks/infra-repo-issue-test-1/issues/22

Some psets are new and the facility part type enum has not been merged yet. Those psets will need some additional love.

### CODE REVIEW

File: IFC4x3/PropertyConstants/u/UNSET_1n8r4XYi950Rj6LjJ0X9h_.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="UNSET_1n8r4XYi950Rj6LjJ0X9h_" Name="UNSET" UniqueId="71235121-8ac2-4501-bb46-56d4c0849afe">
```
Comment (larswik) [inline IFC4x3/PropertyConstants/u/UNSET_1n8r4XYi950Rj6LjJ0X9h_.xml:2]: Is there a need for a new UNSET constant? Why not NOTDEFINED?

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcPortsAndWaterwaysDomain/PropertySets/Pset_GateHeadCommon/DocPropertySet.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_GateHeadCommon" UniqueId="8dbb748d-b6f3-4559-8587-0575f097df30" ApplicableType="IfcFacilityPart/IfcMarinePartTypeEnum(.GATEHEAD.)" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_GateHeadCommon" UniqueId="8dbb748d-b6f3-4559-8587-0575f097df30" ApplicableType="IfcFacilityPart/IfcMarinePartTypeEnum(.GATEHEAD.)" PropertySetType="PSET_OCCURRENCEDRIVEN">
```
Comment (larswik) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcPortsAndWaterwaysDomain/PropertySets/Pset_GateHeadCommon/DocPropertySet.xml:2]: This ApplicableType-string would probably change to "IfcMarineFacilityPart/GATEHEAD" after the redesign of facility part structure.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcPortsAndWaterwaysDomain/PropertySets/Pset_MarinePartChamberCommon/DocPropertySet.xml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_ChamberCommon" UniqueId="63a907b1-a065-46f5-adbe-1bac26deb245" ApplicableType="IfcFacilityPart/IfcMarinePartTypeEnum(.CHAMBER.)" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_MarinePartChamberCommon" UniqueId="63a907b1-a065-46f5-adbe-1bac26deb245" ApplicableType="IfcFacilityPart/IfcMarinePartTypeEnum(.CHAMBER.)" PropertySetType="PSET_OCCURRENCEDRIVEN">
```
Comment (larswik) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcPortsAndWaterwaysDomain/PropertySets/Pset_MarinePartChamberCommon/DocPropertySet.xml:2]: This ApplicableType-string would probably change to "IfcMarineFacilityPart/CHAMBER" after the redesign of facility part structure.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_RailwayLevelCrossing/DocPropertySet.xml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_RailwayLevelCrossing" UniqueId="7f547f91-6dd0-161c-9824-1f608dd660d9" ApplicableType="IfcFacilityPart/IfcFacilityPartCommonTypeEnum(.LEVELCROSSING.)" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_RailwayLevelCrossing" UniqueId="7f547f91-6dd0-161c-9824-1f608dd660d9" ApplicableType="IfcFacilityPart/IfcFacilityPartCommonTypeEnum(.LEVELCROSSING.)" PropertySetType="PSET_OCCURRENCEDRIVEN">
```
Comment (larswik) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_RailwayLevelCrossing/DocPropertySet.xml:2]: This ApplicableType-string would probably change after the redesign of facility part structure.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_RailwayTrackStructurePart/DocPropertySet.xml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_RailwayTrackStructurePart" UniqueId="011923c1-fbf2-84f9-6215-e3ea43caebc9" ApplicableType="IfcFacilityPart/IfcRailwayPartTypeEnum(.TRACKSTRUCTURE.),IfcFacilityPart/IfcRailwayPartTypeEnum(.PLAINTRACKSUPESTRUCTURE.),IfcFacilityPart/IfcRailwayPartTypeEnum(.TURNOUTSUPERSTRUCTURE.),IfcFacilityPart/IfcRailwayPartTypeEnum(.DILATATIONSUPERSTRUCTURE.)" PropertySetType="PSET_TYPEDRIVENOVERRIDE">
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_RailwayTrackStructurePart" UniqueId="011923c1-fbf2-84f9-6215-e3ea43caebc9" ApplicableType="IfcFacilityPart/IfcRailwayPartTypeEnum(.TRACKSTRUCTURE.),IfcFacilityPart/IfcRailwayPartTypeEnum(.PLAINTRACKSUPESTRUCTURE.),IfcFacilityPart/IfcRailwayPartTypeEnum(.TURNOUTSUPERSTRUCTURE.),IfcFacilityPart/IfcRailwayPartTypeEnum(.DILATATIONSUPERSTRUCTURE.)" PropertySetType="PSET_OCCURRENCEDRIVEN">
```
Comment (larswik) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_RailwayTrackStructurePart/DocPropertySet.xml:2]: These ApplicableType-strings would probably change after the redesign of facility part structure.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoadDomain/PropertySets/Pset_RoadDesignCriteriaCommon/DocPropertySet.xml [hunk 6]
```diff
@@ -1,5 +1,5 @@
...
+<DocPropertySet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="Pset_RoadDesignCriteriaCommon" UniqueId="3c1864c7-6e95-46c7-a52f-5edba67bfffb" ApplicableType="IfcRoad,IfcFacilityPart/IfcRoadPartTypeEnum(.ROADSEGMENT.),IfcFacilityPart/IfcFacilityPartCommonTypeEnum(.SEGMENT.),IfcFacilityPart/IfcFacilityPartCommonTypeEnum(.JUNCTION.),IfcFacilityPart/IfcFacilityPartCommonTypeEnum(.LEVELCROSSING.),IfcFacilityPart/IfcRoadPartTypeEnum(.RAILWAYCROSSING.),IfcFacilityPart/IfcRoadPartTypeEnum(.PEDESTRIAN_CROSSING.),IfcFacilityPart/IfcRoadPartTypeEnum(.BICYCLECROSSING.),IfcFacilityPart/IfcRoadPartTypeEnum(.ROUNDABOUT.),IfcFacilityPart/IfcRoadPartTypeEnum(.INTERSECTION.),IfcFacilityPart/IfcRoadPartTypeEnum(.TOLLPLAZA.)" PropertySetType="PSET_OCCURRENCEDRIVEN">
```
Comment (larswik) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRoadDomain/PropertySets/Pset_RoadDesignCriteriaCommon/DocPropertySet.xml:2]: These ApplicableType-strings would probably change after the redesign of facility part structure.

**Review by larswik — COMMENTED** [review]
A few comments regarding ApplicableType-strings and the redesign of facility part structure.

**Review by pjanck — APPROVED** [review]

**Comment by pjanck** [comment]
Except for Lars comments, of course :P

**Review by anborr — APPROVED** [review]
Check ApplicableTypes for consitency with new IfcFacilityPartType

File: IFC4x3/PropertyConstants/u/UNSET_1n8r4XYi950Rj6LjJ0X9h_.xml (hunk 1 above)
Comment (SergejMuhic) [reply IFC4x3/PropertyConstants/u/UNSET_1n8r4XYi950Rj6LjJ0X9h_.xml:2]: To address with @aothms

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcPortsAndWaterwaysDomain/PropertySets/Pset_GateHeadCommon/DocPropertySet.xml (hunk 2 above)
Comment (SergejMuhic) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcPortsAndWaterwaysDomain/PropertySets/Pset_GateHeadCommon/DocPropertySet.xml:2]: Yes, indeed. But we are handling psets in a separate PR, so let us leave it.

**Review by larswik — APPROVED** [review]
UNSET was fine!
Pset applicabilities handled in another PR.

**Pull request merged**